gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -281,13 +294,11 @@ public void run() {
                   if (checksumsEnabled) {
                     Hasher chunkHasher = Hashing.crc32c().newHasher();
                     for (ByteBuffer buffer : chunkData.asReadOnlyByteBufferList()) {
-                      chunkHasher.putBytes(buffer);
-                    }
-                    for (ByteBuffer buffer : chunkData.asReadOnlyByteBufferList()) {
+                      chunkHasher.putBytes(buffer.duplicate());
```

Comment:
Why do we need `duplicate` call?

Reply:
becuz putBytes actually modify the putBytes. That's also why it was in 2 for loops originally. Just figured this way might be clearer.